### ISSUE TITLE
Backend: Vet Centers are not listed as "VA facilities" when adding an Event location

### ISSUE THREAD

**Issue by stefaniefgray** [department-of-veterans-affairs/va.gov-cms#9483]
## Describe the defect
As a VAMC Editor, I would like to add an Event to my site that is hosted at a local Vet Center. (Real life example: VA Bedford is hosting an Event at the Lowell Vet Center.)

When adding the Event location and selecting "At a VA Facility," I expect to see Vet Centers listed within the VA facility location dropdown menu. 

However, there are none listed. This does not make any sense, because Vet Centers are clearly VA facilities.

## To Reproduce

1. While adding or editing a VAMC Event page, select **Location type = At a VA facility**
Link: https://prod.cms.va.gov/node/add/event

2. If you start typing a sample word such as “North” within the “Facility Location” field (see screenshot below), the names of various VAMCs will appear within the dropdown menu that follows.

3. However, if you start typing “Vet Center,” there are zero Vet Centers listed. (See screenshot below.)

## Expected behavior
Vet Centers are VA facilities and should therefore be included within the list of possible VA facility locations for Events.

### **Here is what this menu looks like under normal circumstances after typing the word "North":**

![Screen Shot 2022-06-15 at 8 42 43 PM](https://user-images.githubusercontent.com/9059428/173980114-faa13e20-c299-4198-bf9e-ea31c7cd86e7.png)

### **However, if you type "Vet Center," there are no Vet Centers listed within this menu:**

![Screen Shot 2022-06-15 at 12 57 56 PM](https://user-images.githubusercontent.com/9059428/173980312-cd5f3d43-aea4-46df-9bc0-6d1234da91ab.png)

## Tasks
Expose Vet Centers (content type) in the "Where should the event be listed?" field

## Assumptions
1. Events at a Vet Center location must publish to an existing Event list, e.g. for the greater VAMC. Vet Center pages do not show Event lists.
2. Mobile Vet Centers move location, so events related to a Mobile Vet Center should be listed as "At a non-VA location" and address listed where the Mobile Vet Center will be for the event.
3. FE will adapt seamlessly.

## Tasks
- catch up on the how
- confirm assumptions above
- update View
- QA in tugboat
- code review (1 person)
- FE investigation will be handled in another ticket (if needed)

## Acceptance criteria
- [ ] Only Facilities with published nodes are included in Facility Location list
- [x] Vet Center - Mobile Vet Center is **not** included in Facility location list 
- All other facility types in the CMS should be available in the Facility Location select list, including:
  - [x] Vet Center
  - [x] Vet Center - Community Access Point
  - [x] Vet Center - Outstation
  - [ ] NCA Facility
  - [x] VAMC Facility
  - [ ] VBA Facility
- [ ] Include @swirtSJW as code reviewer on your PR
- [ ] Test the FE against the BE changes (consultation with FE engineer baked-in)
 

 ## Additional context
Original Jira ticket: https://va-gov.atlassian.net/browse/VAHELP-4350
Slack thread: https://dsva.slack.com/archives/C0FQSS30V/p1655320252428059

## Labels
(You can delete this section once it's complete)
- [x] Issue type (red) (defaults to "Defect")
- [ ] CMS subsystem (green)
- [ ] CMS practice area (blue)
- [x] CMS workstream (orange) (not needed for bug tickets)
- [ ] CMS-supported product (black)

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public Websites`
- [x] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by jilladams** [comment]
@kevwalsh @dsinla flagging: we may have Drupal capacity in this sprint / next sprint, if this is something PW could help with, since we have some Events crossover.

**Comment by jilladams** [comment]
@davidconlon Before we take action here, we'd like to confirm: Vet Centers should show as options for a VA Facility to associate with an Event? 

I know Vet Centers are newish in terms of CMS, so just want to make sure we aren't getting ahead of ourselves if we have capacity to go ahead with this change.

**Comment by davidconlon** [comment]
Yes. If it's a facility in the CMS it should be available as a VA location to select from. 
This includes all types. VBA, VHA incl. Vet Centers, NCA.

**Comment by jilladams** [comment]
@swirtSJW noted that Vet Centers do not yet have events. Before we do this work, we need to confirm the implications there. 
1. Would it do harm to expose VCs as a Facility you can associate with an Event? 
2. Are VC Facility templates updated / do they exist such that those events would show somewhere? (e.g. what's the front end lift to make that VC / Event data visible to veterans?)

etc.

**Comment by jilladams** [comment]
From Slack: 

1. for an event to show to Veterans, the event must appear on an [Event list](https://prod.cms.va.gov/admin/content?title=&type=event_listing&moderation_state=All&owner=All). 
2. Any facility that wishes to show an events must create an Event list for that facility.
3. The template must be configured to show up in the right place. e.g. if a Vet Center creates an Event list, we need to confirm that the templates are configured correctly to show that Event list on that facility page. TBD the FE lift for this. 
4. And, this ticket, we have to expose those Facilities to be selectable as the Event location. We can do this work, allow editors to create an Event for a Vet Center location, but that Event will not show to veterans without steps 1-3.

**Comment by jilladams** [comment]
Per Dave sync: Vet Center events have other concerns, including but not limited to node count impact on publishing times in a world without Accelerated publishing, e.g. 300 Vet Centers generating event, etc. content. 

For now, scope of ticket should be: 
1. Expose Vet Centers in the "Where should the event be listed?" field
2. Events at a Vet Center location must publish to an existing Event list, e.g. for the greater VAMC.

**Comment by chri5tia** [comment]
My discovery notes

Shortcuts:
Event list content type:
/admin/structure/types/manage/event_listing/fields (see `field_office`)

Field office field settings only include VAMC
https://va-gov-cms.ddev.site/admin/structure/types/manage/event_listing/fields/node.event_listing.field_office

View:
admin/structure/views/view/content_entity_reference_source

View uses `field_office: content` relationship.
Does appear in vet centers: `Appears in: health_care_region_detail_page, support_service, person_profile, event_listing, publication_listing, vamc_operating_status_and_alerts, story_listing, press_releases_listing, locations_listing, health_services_listing, leadership_listing, vamc_system_policies_page, vet_center_outstation, vet_center_mobile_vet_center, vet_center_locations_list, vet_center_facility_health_servi, vet_center_cap, vamc_system_register_for_care, vamc_system_medical_records_offi, vamc_system_billing_insurance.`

**Comment by chri5tia** [comment]
can't upload the screenshot for some reason so here's a diff

```
   config:
     - field.storage.node.field_office
     - node.type.event_listing
+    - node.type.health_care_local_health_service
     - node.type.health_care_region_page
     - node.type.office
+    - node.type.vba_facility
+    - node.type.vet_center
+    - node.type.vet_center_cap
+    - node.type.vet_center_facility_health_servi
+    - node.type.vet_center_locations_list
+    - node.type.vet_center_mobile_vet_center
+    - node.type.vet_center_outstation
+    - node.type.vha_facility_nonclinical_service
```

**Comment by chri5tia** [comment]
I think my PR is going to only include these changes. Putting this here in case I make changes... disregard.

```
diff --git a/config/sync/views.view.content_entity_reference_source.yml b/config/sync/views.view.content_entity_reference_source.yml
index 40721840..422d3eb8 100644
--- a/config/sync/views.view.content_entity_reference_source.yml
+++ b/config/sync/views.view.content_entity_reference_source.yml
@@ -6,11 +6,16 @@ dependencies:
     - field.storage.node.field_administration
     - node.type.event_listing
     - node.type.health_care_local_facility
+    - node.type.nca_facility
     - node.type.office
     - node.type.person_profile
     - node.type.press_releases_listing
     - node.type.publication_listing
     - node.type.story_listing
+    - node.type.vba_facility
+    - node.type.vet_center
+    - node.type.vet_center_cap
+    - node.type.vet_center_outstation
   module:
     - node
     - user
@@ -307,6 +312,12 @@ display:
           operator: in
           value:
             event_listing: event_listing
+            nca_facility: nca_facility
+            health_care_local_facility: health_care_local_facility
+            vba_facility: vba_facility
+            vet_center: vet_center
+            vet_center_cap: vet_center_cap
+            vet_center_outstation: vet_center_outstation
           group: 1
           exposed: false
           expose:
```

**Comment by chri5tia** [comment]
I'm not clear on the difference, in the story above, between "Where should this event be listed" and "Location type". I see that "Location type" is where the change needs to occur, but curious about the relationship and difference between these two.

Label "Where should the event be listed" = `field_listing` = entity reference view: event listing
(This is where I proposed the above changes; rethinking that now.)

Label: "Facility location" = `field_facility_location ` = entity reference = entity reference view: systems > filtered by content type: VAMC
Adding requested content types **here**, as follows:

 Vet Center
 Vet Center - Community Access Point
 Vet Center - Outstation
 NCA Facility
 VAMC Facility
 VBA Facility

**Comment by chri5tia** [comment]
![Screen Shot 2022-08-19 at 2 06 16 AM](https://user-images.githubusercontent.com/22764938/185585401-fe3fbc70-d65d-4412-82a5-548fe651e982.png)

**Comment by jilladams** [comment]
Note from PR review on #10357 :
Allowing editors to specify VBA and NCA locations that have no FE presence means that Events on the front end present with an empty "Where" field. Follow up work required likely includes: 
* Drupal: 
  * verify if we have VBA / NCA address data stored on these facilities anywhere.  If not, re-adding them as an option here is probably a no-op.
  * if so: re-add VBA & NCA to Facility Locations list
  * check GraphQL for any updates needed to pass VBA/NCA address data to front-end
* Front-end: 
  * Update FE logic to present an address for VBA / NCA facilities

(draft from Jill angle, but will update for reality after scrub with Christia this afternoon)

**Comment by jilladams** [comment]
Also note from PR: scope now includes a change to only show Facilities with published nodes in the Facility Location list. Added to ACs.

Additional context: 
https://github.com/department-of-veterans-affairs/va.gov-cms/pull/10357#issuecomment-1226389252
https://github.com/department-of-veterans-affairs/va.gov-cms/pull/10357#issuecomment-1226735070

**Comment by jilladams** [comment]
New info: Vet Centers also have a similar problem, which requires front-end work. CMS work is basically complete and this ticket becomes a back-end specific ticket. What needs to happen (braindump): 
* Front-end ticket 
* GraphQL query info is added on the CMS PR
* CMS changes are already in a Draft PR, so front-end could theoretically use that branch for content-build changes to the template -- need to confirm this with a front-end person
* We need to understand the proper order of operations for getting both CMS & content-build PRs merged, in this case, since this isn't a new field that can be hid, but is changes to an existing field. We _think_ the right answer is: 
  * CMS PR is ready
  * front-end PRs are staged / tested using the CMS branch
  * CMS & FE meet together when both PRs are ready, for synchronous test / confirmation & then both merge.

**Comment by jilladams** [comment]
@chri5tia front-end PR is merged, so backend is unblocked.

**Comment by stefaniefgray** [comment]
THANK YOU @chri5tia @jilladams @swirtSJW !!!
